File: Leetcode/94.go
package Leetcode

// TreeNode Definition for a binary tree node.
type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func inorderTraversal(root *TreeNode) []int {
	ans := make([]int, 0)

	dfs(root, &ans)

	return ans
}

func dfs(node *TreeNode, curr *[]int) {

	if node == nil {
		return
	}

	dfs(node.Left, curr)
	*curr = append(*curr, node.Val)
	dfs(node.Right, curr)

}

File: Leetcode/94.py
from typing import Optional, List


# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        ans: list[int] = []

        def dfs(node: Optional[TreeNode]):
            nonlocal ans
            if not node:
                return

            dfs(node.left)
            ans.append(node.val)
            dfs(node.right)

        dfs(root)
        return ans
